programs/stake/src/stake_state.rs

```diff
@@ -1023,6 +1057,34 @@ impl MergeKind {
     }
 }
 
+/// Calculate the effective credits observed for two stakes whenn merging
+///
+/// When merging two activating or active stakes, the credits observed of the
+/// merged stake is the weighted average of the two stakes' credits observed.
+fn stake_weighted_credits_observed(
+    stake: &Stake,
+    merge_lamports: u64,
+    merge_credits_observed: u64,
+) -> Option<u64> {
+    if stake.credits_observed == merge_credits_observed {
+        Some(stake.credits_observed)
+    } else {
+        let total_stake = stake.delegation.stake.checked_add(merge_lamports)? as u128;
```

Comment:
Same. I _think_ the compiler will elide the panic checks when it can determine that the conversion can take place without loss of precision, but no harm in being explicit

Reply:
Will do.  Mathematically this *shouldn't* happen, since we're averaging two `u64` values, but it'll certainly protect us in case anything is wrong